src/DotVVM.Framework.Tests/Runtime/CapabilityPropertyTests.cs

```diff
@@ -0,0 +1,122 @@
+        [DataRow(typeof(TestControl2), "HtmlCapability;HtmlCapability;TestCapability")]
+        [DataRow(typeof(TestControl3), "HtmlCapability;TestCapability;TestNestedCapability")]
+        [DataRow(typeof(TestControl4), "HtmlCapability;TestCapability;TestNestedCapability")]
+        [DataRow(typeof(HtmlGenericControl), "HtmlCapability")]
+        [DataRow(typeof(Button), "HtmlCapability;TextOrContentCapability")]
+        public void ControlRegisteredCapabilities(Type control, string capabilities)
+        {
+            var c = DotvvmCapabilityProperty.GetCapabilities(control).Select(c => c.PropertyType.Name).ToArray();
+            Array.Sort(c);
+
+            Assert.AreEqual(
+                capabilities,
+                string.Join(";", c)
+            );
+        }
+
+        public class TestControl1:
+            HtmlGenericControl,
+            IObjectWithCapability<HtmlCapability>,
+            IObjectWithCapability<TestCapability>
+        {
+            public string Something
+            {
+                get { return (string)GetValue(SomethingProperty); }
+                set { SetValue(SomethingProperty, value); }
+            }
+            public static readonly DotvvmProperty SomethingProperty =
+                DotvvmProperty.Register<string, TestControl1>(nameof(Something), "blbost");
+        }
+
+        public class TestControl2:
+            HtmlGenericControl,
+            IObjectWithCapability<TestCapability>
+        {
+            public string Something
+            {
+                get { return (string)GetValue(SomethingProperty); }
+                set { SetValue(SomethingProperty, value); }
+            }
+            public static readonly DotvvmProperty SomethingProperty =
+                DotvvmProperty.Register<string, TestControl2>(nameof(Something), "blbost2");            
+
+            public static readonly DotvvmProperty AnotherHtmlCapabilityProperty =
+                DotvvmCapabilityProperty.RegisterCapability<HtmlCapability, TestControl2>(nameof(AnotherHtmlCapabilityProperty), "AnotherHtml:");
+        }
+
+        public class TestControl3:
+            DotvvmControl,
+            IObjectWithCapability<TestNestedCapability>
+        {       
+            public static readonly DotvvmProperty TestCapabilityProperty =
+                DotvvmCapabilityProperty.RegisterCapability<TestNestedCapability, TestControl3>(nameof(TestCapabilityProperty));
+        }
+        public class TestControl4:
+            HtmlGenericControl,
+            IObjectWithCapability<TestNestedCapability>
+        {       
+        }
+
+        [DotvvmControlCapability]
+        public sealed class TestCapability
+        {
+            public string Something { get; set; } = "kokosovina";
+            public string SomethingElse { get; set; } = "baf";
+        }
+
+        [DotvvmControlCapability]
+        public sealed class TestNestedCapability
+        {
+            public string Something { get; set; } = "abc";
```

Comment:
TODO: test for case insensitivity

Reply:
Turns out, we actually allow two DotVVM properties to have the same name with different casing... It only crashes the compiler later :facepalm: I will not change it in this PR, so we can review it separately. Also, I'm not sure what is the best solution:

* we can make the registration case insensitive, so it will throw an error when it's registered
* we can just adjust the error message compiler throws


Not sure whether the DotvvmProperty.Resolve method should case sensitive or not.